app/models/smart_proxy.rb

```diff
@@ -111,15 +111,15 @@ def sanitize_url
   end
 
   def associate_features
-    return true if Rails.env == 'test'
```

Comment:
It looks like removing this is affecting a very large number of tests - sorry, the check might have to remain.

I do have one more suggestion if you'd like to continue with it - separate the feature association from the validation. So this method with the ProxyAPI::Features call becomes a `before_validation` and is skipped when Rails.env.test?, but the validation that the smart proxy has at least one feature is a normal validator which can be tested in isolation from the feature assignment.